File: Week 4 - Strings and Methods/Lab 4/Tutor Demo ZWordCount/ZWordCount.java
public class ZWordCount {
    public static void main(String[] args) {
        String[] sentences = {
            "green ideas Zleep furiously",
            "a zebra graZed",
            "zorro WAZ zore",
            "I feel snoozy"
        };
        for (String sentence : sentences) {
            System.out.println(sentence);
        }
        System.out.println("Words containing a 'z' = " + zWords(sentences));
    }

    // for-each loop
    // sum
    public static int zWords(String[] sentences) {
        int sum = 0;
        for (String sentence: sentences) {
            sum += zWords(sentence.toLowerCase());
        }
        return sum;
    }

    // split the sentence into words 
    // for-each loop
    public static int zWords(String sentence) {
        int count = 0;
        String[] words = sentence.split(" ");
        for (String word: words) {
            if (containsZ(word)) {
                count++;
            }
        }
        return count;
    }

    // return boolean if word contains z
    public static boolean containsZ(String word) {
        return word.toLowerCase().contains("z");
    }
}

File: Week 4 - Strings and Methods/Lab 4/Tutor Demo ZWordCount/ZWordCount.py
def containsZ(word):
    return "z" in word.lower()


def zWord(sentence):
    count = 0
    sentence = sentence.split()
    for words in sentence:
        if containsZ(words):
            count = count + 1

    return count


def zWords(sentences):
    sum = 0
    for sentence in sentences:
        sum = sum + zWord(sentence.lower())
    return sum


def output(sentences):
    return "Words containing a 'z' = " + str(zWords(sentences))


if __name__ == "__main__":
    sentences = ["green ideas Zleep furiously", "a zebra graZed", "zorro WAZ zore", "I feel snoozy"]
    for sentence in sentences:
        print(sentence)
    print(output(sentences))

